Added `PrepareProposal` and `ProcessProposal` data `CHANGELOG_PENDING.md`. Reworked `UPGRADING.md` (#9315)
tendermint/tendermint: UPGRADING.md

### ABCI Changes

* The `ABCIVersion` is now `0.18.0`.

* Added new ABCI methods `PrepareProposal` and `ProcessProposal`. For details,
  please see the [spec](spec/abci/README.md). Applications upgrading to
  v0.37.0 must implement these methods, at the very minimum, as described
  [here](spec/abci/apps.md)
* Deduplicated `ConsensusParams` and `BlockParams`.
  In the v0.34 branch they are defined both in `abci/types.proto` and `types/params.proto`.
  The definitions in `abci/types.proto` have been removed.

>   This in process applications should make sure they are not using the deleted

Reviewer: Did you perhaps mean something like:

```suggestion
  In-process applications should make sure they are not using the deleted
```

  version of those structures.
* In v0.34, messages on the wire used to be length-delimited with `int64` varint
  values, which was inconsistent with the `uint64` varint length delimiters used
  in the P2P layer. Both now consistently use `uint64` varint length delimiters.
* Added `AbciVersion` to `RequestInfo`.

>   Applications should check that Tendermint's ABCI version matces the one they expect

Reviewer: ```suggestion
  Applications should check that Tendermint's ABCI version matches the one they expect
```

  in order to ensure compatibility.

> * The method `SetOption` has been removed from the ABCI.Client interface.
>   This feature was used in the early ABCI implementation's.

Reviewer: Should we not also mention something about how we've deprecated this in the protos, like:

```suggestion
* The `SetOption` method has been removed from the ABCI `Client` interface.
  The corresponding Protobuf types have been deprecated.
```

### Blockchain Protocol

> * `Header#LastResultsHash`, which is the root hash of a Merkle tree built from
> `ResponseDeliverTx(Code, Data)` as of v0.34 also includes `GasWanted` and `GasUsed`
> fields.

Reviewer: ```suggestion
* `Header#LastResultsHash`, which is the root hash of a Merkle tree built from
  `ResponseDeliverTx(Code, Data)` as of v0.34 also includes `GasWanted` and `GasUsed`
  fields.
```

* The `Verifier` was broken up into two pieces:
    * Core verification logic (pure `VerifyX` functions)
    * `Client` object, which represents the complete light client

> * The new light clients stores headers & validator sets as `LightBlock`s

Reviewer: ```suggestion
* The new light client stores headers and validator sets as `LightBlock`s
```
